plugins/amqp_compressed_proof_plugin/amqp_compressed_proof_plugin.cpp

```diff
@@ -0,0 +1,391 @@
+      }));
+      controller_connections.emplace_back(controller.accepted_block.connect([this,&controller](const chain::block_state_ptr& p) {
+         on_accepted_block(p, controller);
+      }));
+      controller_connections.emplace_back(controller.block_start.connect([this](uint32_t block_num) {
+         on_block_start(block_num);
+      }));
+   }
+
+   void on_applied_transaction(const chain::transaction_trace_ptr& trace) {
+      if(!trace->receipt)
+         return;
+      //only executed & delayed transaction traces would make it in to action_mroot
+      if(trace->receipt->status != chain::transaction_receipt::status_enum::executed &&
+         trace->receipt->status != chain::transaction_receipt::status_enum::delayed)
+         return;
+
+      if(chain::is_onblock(*trace))
+         onblock_trace.emplace(trace);
+      else
+         cached_traces[trace->id] = trace;
+   }
+
+   void on_accepted_block(const chain::block_state_ptr& bsp, const chain::controller& controller) {
+      std::set<action_entry> action_entries_this_block;
+
+      if(onblock_trace) {
+         for(const chain::action_trace& at : (*onblock_trace)->action_traces)
+            action_entries_this_block.emplace(at);
+      }
+      for(const chain::transaction_receipt& r : bsp->block->transactions) {
+         transaction_id_type id;
+         if(r.trx.contains<chain::transaction_id_type>())
+            id = r.trx.get<chain::transaction_id_type>();
+         else
+            id = r.trx.get<chain::packed_transaction>().id();
+         auto it = cached_traces.find(id);
+         EOS_ASSERT(it != cached_traces.end(), chain::misc_exception, "missing trace for transaction ${id}", ("id", id));
+         for(const chain::action_trace& at : it->second->action_traces)
+            action_entries_this_block.emplace(at);
+      }
+      clear_caches();
+
+      if(irreversible)
+         reversible_action_entries_index.emplace(reversible_action_entries{bsp->block_num, bsp->id, std::move(action_entries_this_block)});
+      else
+         run_callbacks_on_block(bsp, action_entries_this_block, controller);
+   }
+
+   void on_irreversible_block(const chain::block_state_ptr& bsp, const chain::controller& controller) {
+      if(!irreversible)
+         return;
+
+      if(const auto it = reversible_action_entries_index.find(bsp->id); it != reversible_action_entries_index.end()) {
+         const std::set<action_entry>& action_entries = it->action_entries;
+
+         run_callbacks_on_block(bsp, action_entries, controller);
+
+         reversible_action_entries_index.get<by_block_num>().erase(
+                 reversible_action_entries_index.get<by_block_num>().begin(),
+                 reversible_action_entries_index.get<by_block_num>().upper_bound(bsp->block_num)
+         );
+      }
+   }
+
+   void run_callbacks_on_block(const chain::block_state_ptr& bsp, const std::set<action_entry>& action_entries, const chain::controller& controller) {
```

Comment:
It would be nice if there were a way to limit the size of the proof, splitting it into several proofs as needed.

Reply:
Is that something that should be rectified ASAP or can it wait until later?

Reply:
It can wait.